### DISCUSSION TITLE
CFD updates irregularly, now stalled for >7 days and I'm wondering about the future of this feature

### DISCUSSION THREAD

**Discussion by richardolsson** [community/community#61771] [Projects and Issues] [answered]
### Select Topic Area

Bug

### Feature Area

Projects

### Body

I love the new projects and the CFD historical chart under Insights. However, I found it to be quite buggy (or at least working irregularly) in both fundamental and superficial ways, and I'm wondering if that's to be expected. It makes me think it's not getting a lot of love but I would be very happy to be corrected as I would like to see this feature improve and prosper.

The most fundamental issue is that the CFD chart of our private projects tend to update very irregularly. Sometimes it updates immediately when issues change status. Sometimes it instead seems to happen overnight. But in our most recent project, it hasn't updated for more than 7 days, even as we keep working the project and other aspects seem to work fine (e.g. automation).

How often is a project CFD chart _supposed_ to update? What within our control as users affect how often it's updated?

<img width="880" alt="image" src="https://github.com/community/community/assets/550212/c9f8c8ca-2ab8-423c-9e6b-7a68a8bb802f">

Another example that I've seen in my usage of the feature as well as in other users' screenshots is the "Invalid date undefined. J u" label on the details popper when hovering the chart (in screenshot above). I've had that error for as long as I can remember, at least multiple months. That seems like such a simple little mistake that someone would discover and fix quickly, hence my worry that the development of historical charts may not be very highly prioritized?

For further investigation by GitHub staff, the project I'm currently experiencing issues with is project 54 in the [zetkin organization](https://github.com/zetkin).

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by buritos** [comment]
I am experiencing similar issues with project insights. I don't mind it being slightly off, but in the last couple of days the charts won't even load. There is an error message saying "Error Loading data" in the UI. In my browser's dev tools network tab, I see a `503 Service unavailable` response when requesting `POST https://github.com/memexes/<ID>/insights/query?period=2W`. Given how we use project insights on a weekly basis, I'd love to see that fixed :pray:

**Comment by rileybroughten** [comment]
👋 Thank you for opening this discussion and apologies for the errors you have been encountering. Changes are coming to project insights where support for the `Group by` field on historical charts (charts with `Time` as the X-axis) is being phased out over the coming weeks, which may result in some disruptions to your charts in the meantime. You can see the [documentation](https://docs.github.com/en/issues/planning-and-tracking-with-projects/viewing-insights-from-your-project/about-insights-for-projects) for more details. Additionally, we'll be providing an updated "Burn up" chart so you can still view the state of your items over time to continue to track progress for your project items.

**Reply by taylor-lindores-reeves** [reply]
Hi @rileybroughten, I am glad I came across your comment as it looks like we are now seeing the following burn-up chart:

1 month ago:
![Picture 1](https://github.com/community/community/assets/27443324/f2a6d94f-2ef1-4cba-9667-8dd0a45910c8)

Today:
![Screenshot 2023-08-29 at 09 52 52](https://github.com/community/community/assets/27443324/fba691d0-2b9c-4cd1-9554-2d60f489ae22)

As you can see, currently I'm stuck with "open" as the default value, which is unclear and I can glean no meaningful insight from this chart. What are the immediate plans for this?

**Reply by rileybroughten** [reply, marked as answer]
We've updated the Burn up chart and historical charts experience, so it now shows the status changes of your items over time - check out the [changelog note](https://github.blog/changelog/2023-09-07-github-issues-projects-september-7th-update/#bar_chart-updates-to-historical-charts) and the [documentation](https://docs.github.com/en/issues/planning-and-tracking-with-projects/viewing-insights-from-your-project/about-insights-for-projects#about-historical-charts).

As you add issues to your project and those are opened and closed over time, as well as pull requests to show pull requests that are opened, merged, and closed over time, you'll see the chart reflect those states changes. Unfortunately, there is not a way to break it up further by the `Status` field in your project or any custom fields, as `Group by` is no longer supported for historical charts.

**Comment by richardolsson** [comment]
Thank you for your answers @rileybroughten, which helps me understand what has been going on. The new experience seems to be working as intended for us.

Unfortunately, for our team this is not an improvement, but quite the opposite. I'm letting you know in order to give you some insight into what it means for us (and perhaps other users as well), with the utmost respect for your research and whatever reasons you may have to make this change.

The "Group by status" feature was crucial for us to understanding where bottlenecks exist in our process. Right now, looking back at the last couple of weeks in our historical chart indicates that we have made very little progress, when in fact the real story is that we've made a lot of progress (solving bugs and submitting pull requests) but there's been a PR review bottleneck.

As it's happening, this is fine, because we can just look at the current state of our project board and understand what is going on, but towards the end of our release cycle when we look back and analyze our workflow, we will have to remember that the plateau that looks like no progress was made, was actually a period where we failed to review PRs.

If this sounds weird to you, please let me know. Maybe I'm missing something, or maybe our process (four column kanban with todo, in progress, review, done) is not a priority.

**Comment by maxshelley** [comment]
> Unfortunately, for our team this is not an improvement, but quite the opposite. I'm letting you know in order to give you some insight into what it means for us (and perhaps other users as well), with the utmost respect for your research and whatever reasons you may have to make this change.

I would agree with @richardolsson's post here, that this is a regression in the usefulness of these charts and assumes that open/closed are the only relevant/useful states for reviewing progress historically (and as we can't prevent a linked PR from auto-closing an issue, open/closed are not states we have control over when using issues to represent work in progress). @rileybroughten Is there any plan to review the removal of this 'group by' feature?
